server/tilesource/base.py

```diff
@@ -1726,8 +1724,7 @@ def canRead(cls, path, *args, **kwargs):
 
 # Girder specific classes
 
-if girder:
-
+if girder:  # noqa - the whole class is allowed to exceed complexity rules
```

Comment:
This noqa disables all linting for the code block.  Can the logic in `_getLargeImagePath` be simplified to not exceed complexity rules?

Reply:
The logic in `_getLargeImagePath` doesn't exceed complexity rules.  The linter is measuring the complexity of the total `if girder:` block, which is the total complexity of the `GirderTileSource` class and complaining about that.  I'm not sure how to get the linter to stop that short of moving the `GirderTileSource` to a secondary source file which, for conceptual reasons, I'd prefer not to do.

Reply:
It seems kind of confusing that depending on the runtime context this class is either defined,  or not defined.  The behavior i would expect is to see is a `RuntimeError` being raised either when importing or instantiating in a non-valid context.

Reply:
Girder is optional.  Girder insists on having plugins have a server module.  Until we refactor Girder to allow pip installable plugins, there is a mashup of Girder and Girder-free code.  Somewhere in this mashup you need to have conditionally defined classes, as some can only be defined if Girder is present.  A conditional with an import would hide this more, but conceptually it is doing the same thing, and I think such a change would make it less obvious what is going on (but then I prefer fewer source files compared to what some others prefer).

Reply:
As implemented this will result in imports of `GirderTileSource` throwing `ImportError` depending on the run time context.  To me,  it makes more sense to throw a `RuntimeError` that explicitly notifies the developer what about the run time context is invalid.

Reply:
I think this *should* be an `ImportError` and *not* a `RuntimeError`, as the failure in this condition is that you can't import girder (`RuntimeErrors` are for errors that don't fall into other categories, and this falls into a failure to import).  Admittedly, the error message would be better as `ImportError: no module named girder`, and not `ImportError: cannot import name GirderTileSource` (assumably from something like `from base import GirderTileSource`).

While this is a fine point to discuss, it doesn't actually have anything to do with the current PR, and will probably change (hopefully to @kotific's satisfaction) when issue #232 is resolved.

Reply:
Yeah,  not looking to bike shed this,  sorry to derail.